lib/fluent/plugin/filter_grep.rb

```diff
@@ -54,65 +59,157 @@ def initialize
       config_param :key, :string
       desc "The regular expression."
       config_param :pattern do |value|
-        if value.start_with?("/") and value.end_with("/")
+        if value.start_with?("/") and value.end_with?("/")
           Regexp.compile(value[1..-2])
         else
           Regexp.compile(value)
         end
       end
     end
 
-    # for test
-    attr_reader :_regexps
-    attr_reader :_excludes
+    config_section :and, param_name: :and_conditions, multi: true do
+      config_section :regexp, param_name: :regexps, multi: true do
+        desc "The field name to which the regular expression is applied."
+        config_param :key, :string
+        desc "The regular expression."
+        config_param :pattern do |value|
+          if value.start_with?("/") and value.end_with?("/")
+            Regexp.compile(value[1..-2])
+          else
+            Regexp.compile(value)
+          end
+        end
+      end
+      config_section :exclude, param_name: :excludes, multi: true do
+        desc "The field name to which the regular expression is applied."
+        config_param :key, :string
+        desc "The regular expression."
+        config_param :pattern do |value|
+          if value.start_with?("/") and value.end_with?("/")
+            Regexp.compile(value[1..-2])
+          else
+            Regexp.compile(value)
+          end
+        end
+      end
+    end
+
+    config_section :or, param_name: :or_conditions, multi: true do
+      config_section :regexp, param_name: :regexps, multi: true do
+        desc "The field name to which the regular expression is applied."
+        config_param :key, :string
+        desc "The regular expression."
+        config_param :pattern do |value|
+          if value.start_with?("/") and value.end_with?("/")
+            Regexp.compile(value[1..-2])
+          else
+            Regexp.compile(value)
+          end
+        end
+      end
+      config_section :exclude, param_name: :excludes, multi: true do
+        desc "The field name to which the regular expression is applied."
+        config_param :key, :string
+        desc "The regular expression."
+        config_param :pattern do |value|
+          if value.start_with?("/") and value.end_with?("/")
+            Regexp.compile(value[1..-2])
+          else
+            Regexp.compile(value)
+          end
+        end
+      end
+    end
 
     def configure(conf)
       super
 
-      rs = {}
+      _regexp_and_conditions = {}
```

Comment:
This name is conflicts with attr_reader's definition. Yeah, I know local variable is preferred but using other name , e.g. `_regexp_and_conds`, is better. Others are same.